signer/jws.go

```diff
@@ -0,0 +1,480 @@
+package signer
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+)
+
+const (
+	JwsJsonMediaType SignatureMediaType = "application/jose+json"
+)
+
+const (
+	headerKeyExpiry      = "io.cncf.notary.expiry"
+	headerKeySigningTime = "io.cncf.notary.signingTime"
+	headerKeyCrit        = "crit"
+	headerKeyAlg         = "alg"
+	headerKeyCty         = "cty"
+)
+
+// jwsInternalEnvelope represents the jwsEnvelope-JSON envelope
+type jwsEnvelope struct {
```

Comment:
I wonder why do we need this wrapper around `jwsInternalEnvelope`? We should be able to get the same functionality with just `jwsInternalEnvelope`, right? What am I missing?

Reply:
This will allow us to replace the `jwsInternalEnvelope` with a JWS-JSON library(If we ever find) without having to much refactoring.

Also, doing it this way provides better code organization.

Reply:
Replacing `jwsInternalEnvelope` with JWS-Json library is same effort with or without `jwsEnvelope`. I feel `jwsEnvelope` is an unnecessary abstraction with no real value. Not sure about better code organization but it is making the code hard to read.